pilot/pkg/model/service.go

```diff
@@ -1265,7 +1277,9 @@ func (s *Service) GetAddressForProxy(node *Proxy) string {
 			}
 		}
 
-		if node.Metadata.DNSCapture && node.Metadata.DNSAutoAllocate && s.DefaultAddress == constants.UnspecifiedIP {
+		if node.Metadata.DNSCapture &&
+			(node.Metadata.DNSAutoAllocate || pm.StringBool(features.EnableIPAutoallocate)) &&
```

Comment:
nit: I would rather cast the others to `bool` than this to StringBool. Its just syntactical preference though